### DISCUSSION TITLE
Setting Rate Limiting Policy Using Configuration in YARP

### DISCUSSION THREAD

**Discussion by ProximaB** [dotnet/yarp#2684] [Q&A] [answered]
Is it possible with YARP to set a rate limiting policy using configuration without defining a specific .NET service? I have a requirement to fully configure the rate limiting policy through configuration.

I cannot find information if the YARP support that itself, or it needs to be done on my own.

Here is an example of what I'm trying to achieve:
```json
{
  "RateLimiterPolicy": {
    "FixedWindow": {
      "PermitLimit": 100,
      "QueueLimit": 5
    }
  }
}
```
Thanks in advance for any suggestions.

**Comment by MihaZupan** [comment, marked as answer]
> I cannot find information if the YARP support that itself, or it needs to be done on my own.

No, YARP will only read the `RateLimiterPolicy` as a `string` for the name.
You would have to do the config => policy creation yourself.

Do you need to modify the policy after startup?
